sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1223,37 +1248,36 @@ def task_worker(self):
                         has_update = self.is_cmis_application_update_required(api, host_lanes, host_speed)
                         if not has_update:
                             # No application updates
-                            self.log_notice("{}: READY".format(lport))
+                            self.log_notice("{}: no CMIS application update required...READY".format(lport))
                             self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_READY
                             continue
 
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_DEINIT
+                    elif state == self.CMIS_STATE_DP_DEINIT:
                         # D.2.2 Software Deinitialization
                         api.set_datapath_deinit(host_lanes)
-                        api.set_lpmode(True)
```

Comment:
You are changing generic code. Isn't this set_lpmode is required function?

Reply:
It sounds like this PR will perform CMIS application selection one by one for each datapath(i.e. logical port on the MAC) instead of a one-shot configuration for all the datapatches on a transceiver.

What's unclear to me is as follows:

1. Is there a switch ASIC available to support asynchronous mixed port mode in a port macro? (i.e. Is it possible to have both 1x100G-NZR + 2x100G-PAM4 on a single QSFPDD port, from MAC perspective?)
2. Is there a CMIS transceiver available to support asynchronous mixed port mode in a single module?

In the above scenario, each datapath will require a dedicated clock divisor while they all share the same reference clock source. As far as I know, this is NOT a supported port mode in Broadcom ASIC, but I'm not sure if this is also applicable to other vendors, could you please clarify if there is a such module available for testing? Thank you

Reply:
Xcvr is lisenting to various port change events related to a ports speed, lane mapping etc and these events across various logical ports in the case of breakout port (say one physical port mapping to 4 logical ports) will arrive asynchronously, so we need to handle these events independently, even though it belong to same physical port.